tsconfig.json

```diff
@@ -1,5 +1,8 @@
 {
   "compilerOptions": {
```

Comment:
Consider adding [`exactOptionalPropertyTypes`](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes), which gives the strict optional vs. `undefined` behavior you were talking about. (Maybe in another change?)

Reply:
Yeah, it's time for a some more global cleanups.